service/src/main/java/uk/gov/mca/beacons/api/legacybeacon/application/LegacyBeaconService.java

```diff
@@ -51,4 +54,28 @@ public List<LegacyBeacon> claimByHexIdAndAccountHolderEmail(
 
     return savedLegacyBeacons;
   }
+
+  public void delete(String hexId, String email) {
+    List<LegacyBeacon> legacyBeacons = legacyBeaconRepository.findByHexIdAndOwnerEmail(
+      hexId,
+      email
+    );
+    legacyBeacons.forEach(LegacyBeacon::softDelete);
+
+    LegacyBeacon legacyBeacon = legacyBeacons.get(0);
+    LegacyData legacyBeaconData = legacyBeacon.getData();
+
+    legacyBeaconData.setOwner(new LegacyOwner());
+    legacyBeaconData.setUses(new ArrayList<LegacyUse>());
```

Comment:
Linda said she wanted all the uses removed during the deletion. Also, the existing softDelete for modern beacons removes the uses so I thought another reason to do this would be to ensure consistency across both legacy and modern